lib/src/default_index/composite.rs

```diff
@@ -487,15 +487,19 @@ impl Index for &CompositeIndex {
         Ok(Box::new(self.all_heads()))
     }
 
-    fn heads(&self, candidate_ids: &mut dyn Iterator<Item = &CommitId>) -> Vec<CommitId> {
+    fn heads(
+        &self,
+        candidate_ids: &mut dyn Iterator<Item = &CommitId>,
+    ) -> Result<Vec<CommitId>, RevsetEvaluationError> {
```

Comment:
It seems a little strange to return `RevsetEvaluationError` here because there's no revset visibly involved. Maybe we should introduce a new error like this (basically the same as the `IndexReadError` and `IndexWriteError` errors associated with the `IndexStore` trait):

```
#[derive(Debug, Error)]
#[error(transparent)]
pub struct IndexError(pub Box<dyn std::error::Error + Send + Sync>);
```

Reply:
Added IndexError as you suggested above, thanks!